enhancements/network/no_overlay_mode.md

```diff
@@ -0,0 +1,807 @@
+// field so the default is visible in the CRD schema and applied by the API server automatically.
+// Currently CNO handles the default (treating omitted as Geneve) because the field is feature-gated
+// and existing ungated tests don't expect this field in outputs.
+
+// ovnKubernetesConfig contains the configuration parameters for networks
+// using the ovn-kubernetes network project
+// +openshift:validation:FeatureGateAwareXValidation:featureGate=NoOverlayMode,rule="!has(self.defaultNetworkTransport) || self.defaultNetworkTransport != 'NoOverlay' || has(self.defaultNetworkNoOverlayOptions)",message="defaultNetworkNoOverlayOptions is required when defaultNetworkTransport is NoOverlay"
+// +openshift:validation:FeatureGateAwareXValidation:featureGate=NoOverlayMode,rule="!has(self.defaultNetworkNoOverlayOptions) || self.defaultNetworkNoOverlayOptions.routing != 'Managed' || has(self.bgpManagedConfig)",message="bgpManagedConfig is required when defaultNetworkNoOverlayOptions.routing is Managed"
+// +openshift:validation:FeatureGateAwareXValidation:featureGate=NoOverlayMode,rule="!has(oldSelf.defaultNetworkTransport) || oldSelf.defaultNetworkTransport == '' || has(self.defaultNetworkTransport)",message="defaultNetworkTransport cannot be removed once set to a non-empty value"
+// +openshift:validation:FeatureGateAwareXValidation:featureGate=NoOverlayMode,rule="!has(oldSelf.defaultNetworkNoOverlayOptions) || has(self.defaultNetworkNoOverlayOptions)",message="defaultNetworkNoOverlayOptions cannot be removed once set"
+type OVNKubernetesConfig struct {
+	// ... existing fields ...
+
+	// defaultNetworkTransport describes the transport protocol for east-west traffic for the default network.
+	// Allowed values are "NoOverlay" and "Geneve".
+	// When set to "NoOverlay", the default network operates in no-overlay mode.
+	// When set to "Geneve", the default network uses Geneve overlay.
+	// When omitted, this means the user has no opinion and the platform chooses a reasonable default which is subject to change over time.
+	// The current default is "Geneve".
+	// +openshift:enable:FeatureGate=NoOverlayMode
+	// +kubebuilder:validation:Enum=NoOverlay;Geneve
+	// +kubebuilder:validation:XValidation:rule="oldSelf == '' || self == oldSelf",message="defaultNetworkTransport is immutable once set"
+	// +optional
+	DefaultNetworkTransport TransportOption `json:"defaultNetworkTransport,omitempty"`
+
+	// defaultNetworkNoOverlayOptions contains configuration for no-overlay mode for the default network.
+	// It is required when DefaultNetworkTransport is "NoOverlay".
+	// When omitted, this means the user does not configure no-overlay mode options.
+	// +openshift:enable:FeatureGate=NoOverlayMode
+	// +kubebuilder:validation:XValidation:rule="!oldSelf.hasValue() || self == oldSelf.value()",message="defaultNetworkNoOverlayOptions is immutable once set",optionalOldSelf=true
```

Comment:
Add CEL rules to verify the existence of defaultNetworkNoOverlayOptions based on the value of DefaultNetworkTransport.

Reply:
We also need to validate that `defaultNetworkNoOverlayOptions` is not set when defaultNetworkTransport != 'NoOverlay'.